src/core/lib/iomgr/socket_mutator.cc

```diff
@@ -37,8 +37,19 @@ grpc_socket_mutator* grpc_socket_mutator_ref(grpc_socket_mutator* mutator) {
   return mutator;
 }
 
-bool grpc_socket_mutator_mutate_fd(grpc_socket_mutator* mutator, int fd) {
-  return mutator->vtable->mutate_fd(fd, mutator);
+bool grpc_socket_mutator_mutate_fd(grpc_socket_mutator* mutator, int fd,
+                                   grpc_fd_usage usage) {
+  if (mutator->vtable->mutate_fd_2 != nullptr) {
+    grpc_mutate_socket_info info{fd, usage};
```

Comment:
nit: any reason not to invoke `mutate_fd_2`(fd, usage, mutator)` here instead, and get rid of the `grpc_mutate_socket_info` type?